functions/go/set-image/transformer/images_transformer.go

```diff
@@ -0,0 +1,170 @@
+package transformer
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-image/custom"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-image/image_util"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+// SetImage supports the set-image workflow, it uses Config to parse functionConfig, Transform to change the image
+type SetImage struct {
+	// Image is the desired image
+	Image image_util.Image `json:"image,omitempty" yaml:"image,omitempty"`
+	// ConfigMap keeps the data field that holds image information
+	ConfigMap map[string]string `json:"data,omitempty" yaml:"data,omitempty"`
+	// ONLY for kustomize, AdditionalImageFields is the user supplied fieldspec
+	AdditionalImageFields image_util.FsSlice `json:"additionalImageFields,omitempty" yaml:"additionalImageFields,omitempty"`
+	// context logs each detailed result
+	context *fn.Context
+	// resultCount logs the total count image change
+	resultCount int
+}
+
+// Config transforms the data from ConfigMap to SetImage struct
+func (imagetrans *SetImage) Config() error {
+	for key, val := range imagetrans.ConfigMap {
+		switch key {
+		case "name":
+			imagetrans.Image.Name = val
+		case "newName":
+			imagetrans.Image.NewName = val
+		case "newTag":
+			imagetrans.Image.NewTag = val
+		case "digest":
+			imagetrans.Image.Digest = val
+		default:
+			return fmt.Errorf("SubObject has unmatched field type: `data`")
+		}
+	}
+	return nil
+}
+
+// validateInput validates the inputs passed into via the functionConfig
+func (imagetrans *SetImage) validateInput() error {
+	// TODO: support container name and only one argument input in the next PR
+	if imagetrans.Image.Name == "" {
+		return fmt.Errorf("missing `name` to select the image")
+	}
+	if imagetrans.Image.NewName == "" && imagetrans.Image.NewTag == "" && imagetrans.Image.Digest == "" {
+		return fmt.Errorf("missing `newName`, `newTag`, or `digest`, at least one of the fields should not be empty")
```

Comment:
```suggestion
		return fmt.Errorf("must specify one of `newName`, `newTag`, or `digest`")
```

See the kpt error message style guide: go/kpt-errors-style-guide

Reply:
Thanks a lot for the guide!